malleefowl/custom_workflow.py

```diff
@@ -0,0 +1,569 @@
+    - name : Unique name given to each workflow task
+    - url : Url of the WPS provider
+    - identifier : Identifier of a WPS process
+    - inputs : Array of static input required by the WPS process, each describe by a 2 elements array :
+      - Name of the input
+      - Value of the input
+    - linked_inputs : Array of dynamic input required by the WPS process and obtained by the output of other tasks,
+                      each describe by a 2 elements array :
+      - Name of the input
+      - Provenance of the input, describe by a dict :
+        - task : Name of the task from which this input must come from
+        - output : Name of the task output that will be linked
+        - as_reference : Specify the required form of the input** [True: Expect an URL to the input,
+                                                                   False: Expect the data directly]
+    - progress_range : 2 elements array defining the overall progress range of this task :
+      - Start
+      - End
+
+** The workflow executor is able obviously to assign a reference output to an expected reference input and
+   a data output to an expected data input but will also be able to read the value of a reference output to send the
+   expected data input. However, a data output linked to an expected reference input will yield to an exception.
+
+Exemple:
+{
+        "name": "Subsetting workflow",
+        "tasks": [
+            {
+                "name": "Downloading",
+                "url": "http://localhost:8091/wps",
+                "identifier": "thredds_download",
+                "inputs": [["url", "http://localhost:8083/thredds/catalog/birdhouse/catalog.xml"]],
+                "progress_range": [0, 50]
+            },
+            {
+                "name": "Subsetting",
+                "url": "http://localhost:8093/wps",
+                "identifier": "subset_WFS",
+                "inputs": [["typename", "ADMINBOUNDARIES:canada_admin_boundaries"],
+                           ["featureids", "canada_admin_boundaries.5"],
+                           ["mosaic", "False"]],
+                "linked_inputs": [["resource", { "task": "Downloading",
+                                                 "output": "output",
+                                                 "as_reference": False}],],
+                "progress_range": [50, 100]
+            },
+        ]
+    }
+"""
+
+import sys
+import json
+import urllib2
+from cStringIO import StringIO
+from time import sleep
+
+from owslib.wps import WebProcessingService
+from owslib.wps import ComplexDataInput
+from owslib.wps import printInputOutput
+
+from dispel4py.workflow_graph import WorkflowGraph
+from dispel4py import simple_process
+from dispel4py.core import GenericPE
+
+from malleefowl.workflow import run as run_basic_workflow
+
+# For check_status function
+from owslib.wps import WPSExecuteReader
+from owslib.etree import etree
+
+import logging
+logger = logging.getLogger("PYWPS")
```

Comment:
Can cause problems if a module is not well defined in requirements, will only be detected for certain execution flows...